sumpy/expansion/local.py

```diff
@@ -179,8 +179,110 @@ def evaluate(self, kernel, coeffs, bvec, rscale, sac=None):
 
         return kernel.postprocess_at_target(result, bvec)
 
+    def m2l_global_precompute_nexpr(self, src_expansion):
```

Comment:
This adds extra stuff to `Expansion` interface in a kind-of ad-hoc way. Would you be open to making a new ABC that documents this a bit? As a bonus, maybe this would allow us to merge `E2EFromCSRTranslationInvariant` back into the base, because it'll easily be able to recognize expansions that have this.

Reply:
Yes, that's a good idea.